src/modules/keyboardmanager/test/LoadingAndSavingRemappingTests.cpp

```diff
@@ -0,0 +1,521 @@
+            remapBuffer.push_back(std::make_pair(std::vector<std::variant<DWORD, Shortcut>>({ src2, dest2 }), std::wstring()));
+
+            // Assert that remapping set is valid
+            bool isSuccess = (LoadingAndSavingRemappingHelper::CheckIfRemappingsAreValid(remapBuffer) == KeyboardManagerHelper::ErrorType::NoError);
+            Assert::AreEqual(true, isSuccess);
+        }
+
+        // Test if the CheckIfRemappingsAreValid method is successful when valid remaps are passed
+        TEST_METHOD (CheckIfRemappingsAreValid_ShouldReturnNoError_OnPassingValidRemapsOfAllTypes)
+        {
+            std::vector<std::pair<std::vector<std::variant<DWORD, Shortcut>>, std::wstring>> remapBuffer;
+
+            // Remap Ctrl+V to Ctrl+D, Alt+Tab to A, A to B and B to Win+A
+            Shortcut src1;
+            src1.SetKey(VK_CONTROL);
+            src1.SetKey(0x56);
+            Shortcut dest1;
+            dest1.SetKey(VK_CONTROL);
+            dest1.SetKey(0x44);
+            Shortcut src2;
+            src2.SetKey(VK_MENU);
+            src2.SetKey(VK_TAB);
+            Shortcut dest2;
+            dest2.SetKey(CommonSharedConstants::VK_WIN_BOTH);
+            dest2.SetKey(0x41);
+            remapBuffer.push_back(std::make_pair(std::vector<std::variant<DWORD, Shortcut>>({ src1, dest1 }), std::wstring()));
+            remapBuffer.push_back(std::make_pair(std::vector<std::variant<DWORD, Shortcut>>({ src2, 0x41 }), std::wstring()));
+            remapBuffer.push_back(std::make_pair(std::vector<std::variant<DWORD, Shortcut>>({ 0x41, 0x42 }), std::wstring()));
+            remapBuffer.push_back(std::make_pair(std::vector<std::variant<DWORD, Shortcut>>({ 0x42, dest2 }), std::wstring()));
+
+            // Assert that remapping set is valid
+            bool isSuccess = (LoadingAndSavingRemappingHelper::CheckIfRemappingsAreValid(remapBuffer) == KeyboardManagerHelper::ErrorType::NoError);
+            Assert::AreEqual(true, isSuccess);
+        }
+
+        // Test if the CheckIfRemappingsAreValid method is unsuccessful when remaps with null keys are passed
+        TEST_METHOD (CheckIfRemappingsAreValid_ShouldReturnRemapUnsuccessful_OnPassingRemapsWithNullKeys)
+        {
+            std::vector<std::pair<std::vector<std::variant<DWORD, Shortcut>>, std::wstring>> remapBuffer;
+
+            // Remap A to NULL
+            remapBuffer.push_back(std::make_pair(std::vector<std::variant<DWORD, Shortcut>>({ 0x41, NULL }), std::wstring()));
+
+            // Assert that remapping set is invalid
+            bool isSuccess = (LoadingAndSavingRemappingHelper::CheckIfRemappingsAreValid(remapBuffer) == KeyboardManagerHelper::ErrorType::RemapUnsuccessful);
+            Assert::AreEqual(true, isSuccess);
+        }
+
+        // Test if the CheckIfRemappingsAreValid method is unsuccessful when remaps with invalid shortcuts are passed
+        TEST_METHOD (CheckIfRemappingsAreValid_ShouldReturnRemapUnsuccessfulr_OnPassingRemapsWithNullKeys)
```

Comment:
I'd recommend renaming this ShouldReturnRemapUnsuccessful_OnPassingRemapsWithInvalidShortcut. 

Also the shortcut itself should probably have a unit test so that you don't have to test all the permutations of IsValidShortcut == false.

Reply:
Added tests for IsValidShortcut, CheckRepeatedModifier, GetsKeysFromSelectedIndices and DoKeysOverlap to reduce the number of permutations